### DISCUSSION TITLE
How to create Compound Index?

### DISCUSSION THREAD

**Discussion by lahin31** [prisma/prisma#24585] [Q&A] [answered]
### Question

I wanted to create a Compound Index or Multi-Column Index(together) for email and phone. I tried like this,

```
@@unique([email, phone])
@@index([email, phone], name: "email_phone")
```

Then I saw it created index for email, phone, and some more index created. I would like to know what is the proper way to create a Compound Index.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @lahin31 👋 

To create a compound or multi-column index in Prisma Schema, you can use the `@@index` attribute in your model definition. For example:
```prisma
model User {
  id      Int     @id @default(autoincrement())
  email   String
  phone   String

  @@index([email, phone], name: "email_phone")
}
```
This will create a multi-column index on the email and phone fields with the custom name "email_phone". See this [section](https://www.prisma.io/docs/orm/prisma-schema/data-model/models#defining-an-index) of the documentation.

If you also want to ensure that the combination of email and phone is unique, you can use the `@@unique` attribute:
```prisma
model User {
  id      Int     @id @default(autoincrement())
  email   String
  phone   String

  @@unique([email, phone], name: "unique_email_phone")
  @@index([email, phone], name: "email_phone")
}
```
This will create a unique constraint on the combination of email and phone and also create an index with the name "email_phone". See this [section](https://www.prisma.io/docs/orm/reference/prisma-schema-reference#unique-1) of the documentation.

The compound index `@@index` is an index on multiple columns of a table and does not enforce uniqueness of the combined column values. 

The compound unique index `@@unique` also indexes multiple columns, but in addition to improving query performance, it enforces uniqueness across the combination of the specified columns.

If you just want a composite index and do not care about uniqueness, then using `@@index([email, phone], name: "email_phone")` is the way to go.
